extensions/blocks/anchor-fm/anchor-fm.php

```diff
@@ -0,0 +1,113 @@
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+if ( ! class_exists( 'Jetpack_Podcast_Helper' ) ) {
+	\jetpack_require_lib( 'class-jetpack-podcast-helper' );
+}
+
+/**
+ * Registers Anchor.fm integration for the block editor.
+ */
+function register_block() {
+	Blocks::jetpack_register_block( BLOCK_NAME );
+
+	// Register post_meta for connecting Anchor podcasts with posts.
+	register_post_meta(
+		'post',
+		'jetpack_anchor_podcast',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'string',
+		)
+	);
+	register_post_meta(
+		'post',
+		'jetpack_anchor_episode',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'string',
+		)
+	);
+	register_post_meta(
+		'post',
+		'jetpack_anchor_spotify_show',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'string',
+		)
+	);
+}
+
+/**
+ * Checks URL params to determine the Anchor integration action to perform.
+ */
+function process_anchor_params() {
+	$current_screen = \get_current_screen();
+	// TODO: Replace `$current_screen->is_block_editor()` with `wp_should_load_block_editor_scripts_and_styles()` that is introduced in WP 5.6.
+	if ( method_exists( $current_screen, 'is_block_editor' ) && ! $current_screen->is_block_editor() ) {
+		// Return early if we are not in the block editor.
+		return;
+	}
+
+	$post = get_post();
+	if ( ! $post || ! $post->ID ) {
+		return;
+	}
+
+	// phpcs:disable WordPress.Security.NonceVerification.Recommended
+	$podcast_id       = isset( $_GET['anchor_podcast'] ) ? sanitize_text_field( $_GET['anchor_podcast'] ) : null;
+	$episode_id       = isset( $_GET['anchor_episode'] ) ? sanitize_text_field( $_GET['anchor_episode'] ) : null;
+	$spotify_show_url = isset( $_GET['spotify_show_url'] ) ? esc_url_raw( $_GET['spotify_show_url'] ) : null;
+	// phpcs:enable WordPress.Security.NonceVerification.Recommended
+
+	$data = array();
+
+	if ( ! empty( $podcast_id ) ) {
+		$feed = 'https://anchor.fm/s/' . $podcast_id . '/podcast/rss';
+		$rss  = Jetpack_Podcast_Helper::load_feed( $feed );
+		if ( ! \is_wp_error( $rss ) ) {
+			$data['podcastId'] = $podcast_id;
+			update_post_meta( $post->ID, 'jetpack_anchor_podcast', $podcast_id );
+
+			if ( ! empty( $episode_id ) ) {
+				$track = Jetpack_Podcast_Helper::get_track_data( $feed, $episode_id );
+				if ( ! \is_wp_error( $track ) ) {
+					$data['episodeId'] = $episode_id;
+					$data['track']     = $track;
+					update_post_meta( $post->ID, 'jetpack_anchor_episode', $episode_id );
+				}
+			}
+		}
+	}
+
+	if ( ! empty( $spotify_show_url ) ) {
+		$data['spotifyShowUrl'] = $spotify_show_url;
+		if ( get_post_meta( $post->ID, 'jetpack_anchor_spotify_show', true ) !== $spotify_show_url ) {
+			update_post_meta( $post->ID, 'jetpack_anchor_spotify_show', $spotify_show_url );
+			$data['action'] = 'insert-spotify-badge';
+			$data['image']  = \Jetpack::staticize_subdomain( 'https://wordpress.com/i/spotify-badge.svg' );
```

Comment:
To continue the discussion we had earlier, maybe for a v2 we could look into ways to sideload the image onto the site's media library via `media_sideload_image`?

See https://github.com/Automattic/jetpack/blob/4464989e638dd125f1f689195cc18467f6ba3672/modules%2Fvideopress%2Futility-functions.php#L150-L154 for an example.

Reply:
Is this preferred over loading the image from a CDN for some reason?

Reply:
As a general rule of thumb in the WordPress.org plugin directory, it's often preferred to avoid offloading assets to a third-party (whether it's scripts, images, or something else) when it can be avoided, or when it doesn't add any value to the service.

Offloading assets such as an image can have a few disadvantages:
- The third-party service may not be around forever, or may not serve that image forever.
- This may be seen as an attempt at tracking for some. In this case the block is usable by folks using Jetpack's Offline mode, who did not agree to Jetpack's Terms of service because they did not connect their site to WordPress.com.
- The third-party service may not always be accessible to all readers. As an example, readers in China would not be able to see a wpcom-hosted image.

I wouldn't consider this a blocker though. There are other cases in the Jetpack plugin (like the fallback image in open graph meta tags) where we use a wpcom-hosted image in a feature that can be used without a connction to WordPress.com.

Reply:
https://github.com/Automattic/jetpack/issues/18060